src/Voron/StorageEnvironment.cs

```diff
@@ -163,6 +167,83 @@ public StorageEnvironment(StorageEnvironmentOptions options)
             }
         }
 
+        private void OnCreateDirectory()
+        {
+
+            var userArgs = _options.OnCreateDirectoryExecArguments ?? string.Empty;
+            var args = $"{userArgs} {_options.Type} {_options.DatabaseName} {_options.BasePath} ";
+
+            var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = _options.OnCreateDirectoryExec,
+                    Arguments = args,
+                    UseShellExecute = false,
+                    RedirectStandardOutput = false,
+                    RedirectStandardError = true,
+                    CreateNoWindow = true,
+                    Verb = "runas"
+                }
+            };
+
+            var sw = Stopwatch.StartNew();
+
+            try
+            {
+                process.Start();
+            }
+            catch (Exception e)
+            {
+                throw new InvalidOperationException($"Unable to execute '{_options.OnCreateDirectoryExec} {args}'. Failed to start process.", e);
+            }
+
+            var readErrors = process.StandardError.ReadToEndAsync();
```

Comment:
The problem with this approach is that each error line invokes a new callback asynchronously, and even though the process finishes and my program continues, not all the callbacks have been invoked. So to use this feature I'd have to add synchronization and continue my program only after I get the last error (empty line).
The current way is simpler and I can be sure that when I read the entire stream, all errors will be there for sure, because I waited for the process to exit.